src/StudentOrganizer.Core/Models/Course.cs

```diff
@@ -15,7 +16,43 @@ public class Course : Entity
 		public int Semester { get; protected set; }
 		public IEnumerable<Assignment> Assignments => _assignments;
 
+        public Course(string name, string lecturer, Location location, IsoDayOfWeek day, 
+			LocalTime startTime, LocalTime endTime, int semester)
+        {
+			SetName(name);
+			SetLecturer(lecturer);
+			Location = location;
+			DayOfTheWeek = day;
+			StartTime = startTime;
+			EndTime = endTime;
+			SetSemester(semester);
+        }
 
+		public void SetName(string name)
+        {
+            if (string.IsNullOrWhiteSpace(name))
+            {
+				throw new Exception("Name can not be empty.");
+            }
+			Name = name;
+        }
 
+		public void SetLecturer(string lecturer)
+		{
+			if (string.IsNullOrWhiteSpace(lecturer))
+			{
+				throw new Exception("Lecturer can not be empty.");
+			}
+			Lecturer = lecturer;
+		}
+
+		public void SetSemester(int semester)
+		{
+			if (semester < 0)
+			{
+				throw new Exception("Semester can not be lower than zero.");
+			}
+			Semester = semester;
```

Comment:
Tutaj wsadzić górną granicę semestru? Dodatkowo w całym tym modelu może warto by dodać atrybut "stopień studiów" na których odbywa się kurs, np. czy magisterskie czy inżynierskie. Górna granica dla semestru IMO 12, tyle trwają medyczne - wyżej się nie da.

Reply:
Z granicą górną to był mój pomysł wsumie. Ok medyczne mogą mieć 12, ale nie wiem czy gdzieś na świecie semestry mogą mieć jakieś powalone numerowanie także bym odpuścił. Jak ktoś chce niech se zrobi semestr 50. w aplikacji niczego to nam nie popsuje xd.

Reply:
Hmm magisterkie i inżynierskie też not sure, bo np zaczyna się inżynierka to możliwe, że zmienią się startości i skład grupy też także może ktoś będzie tworzył nową grupę i nazwie ją ... Magisterka ... Ogółem spoko pomysł, ale dajmy go na jakąś listę   to do.